forge/lib/validate.js

```diff
@@ -0,0 +1,7 @@
+module.exports = {
+    isFQDN (string) {
+        const regex = /(?=^.{4,253}$)(^((?!-)[a-zA-Z0-9-]{0,62}[a-zA-Z0-9]\.)+[a-zA-Z]{2,63}\.?$)/
```

Comment:
I'm going to need this regex explaining

Reply:
It boils down to:

- `(?=^.{4,253}\.?$)` total FQDN between 4 and 254 characters
-  `(` - start new group
- `^((?!-)` does not begin with a `-` (otherwise valid)
- `[a-zA-Z0-9-]{1,63}` Up to 1-63 alphanumeric characters plus `-`
- `(?<!-)` not ending in a `-`
- `\.` literal .
- `)+` repeat this group unlimited times 
- `[a-zA-Z]{2,63}` Up to 2-63 alphabetic letters for the TLD

Adapted from https://stackoverflow.com/questions/11809631/fully-qualified-domain-name-validation